### ISSUE TITLE
App looks broken when running on production mode

### ISSUE THREAD

**Issue by alvarezguille** [vaadin/bookstore-starter-flow#55]
Tested with `jetty:run` and with `package` + running tomcat both with `-Dvaadin.productionMode` and `-PproductionMode`
Dark theme is not applied, and some other styling issues as seen on the following screenshot
![image](https://user-images.githubusercontent.com/9820084/47213429-aa97b880-d3a3-11e8-9ac5-a760ec96cd1e.png)

Tested with vaadin 12.0.0.alpha4

**Comment by caalador** [comment]
Only thing missing for me is the table-logo.png that doesn't get served.
Have you tried clearing the browser cache and/or clearing all target folders?

**Comment by SomeoneToIgnore** [comment]
Is it fixed now?

**Comment by caalador** [comment]
No. The logo image is fixed, but the problem is with IE11 and that's not fixed.

**Comment by caalador** [comment]
All I can come up for this is that the css files use var() which is not supported for IE only for Edge.
It gets better if I remove all the `var()` usages from the `shared-styles.css`, but still it's not as it should be.

Perhaps there is a problem with the CSS polyfill for some reason.

**Comment by Legioth** [comment]
Isn't it so that the standard polyfill only applies for styles in `<custom-style>`. You'd have to manually apply the polyfilling logic if you want to use it with standalone css files?

**Comment by caalador** [comment]
That could be. Also I noted that for the `[theme~=dark]` IE gets the colors the wrong way so that color that should be `hsla(214, 96%, 96%, 0.9)` is actually `hsla(213, 41%, 16%, 0.94)` which is closer to the wanted background-color of `hsl(214, 35%, 21%)`

**Comment by caalador** [comment]
Also the theme is broken for Vaadin 10.0.5 also when running IE11

**Comment by caalador** [comment]
Dark theme (nor normal) doesn't work at all in IE11 even when running skeleton-starter.
Also clicking the button throws a huge exception on IE11.

**Comment by caalador** [comment]
For the Dark theme it might be a problem with the bundler that it probably doesn't check for the theme variant dark and only brings in the light lumo theme.

**Comment by caalador** [comment]
The transpilation step leaves me with this when IE11 loads the colors.html
![screen shot 2018-10-29 at 13 38 33](https://user-images.githubusercontent.com/2726090/47647553-fd315b80-db7f-11e8-81d0-1fabfc9f3321.png)

**Comment by jouni** [comment]
Investigating this as well. I can’t make the dark theme work in IE11 in a pure Polymer app at the moment either.

**Comment by jouni** [comment]
Damn, I don’t seem to be able to make style modules work at all (`<style include="foobar">`).

**Comment by jouni** [comment]
Alright, this seems like a fundamental issue in Lumo itself, not only in this starter.

ShadyCSS seems to have a special case for the `html` selector. If you combine that with anything else (like `html[theme~≈"dark"]` or `html, [theme~="dark"]`) it will silently ignore polyfiling any custom properties inside that selector.

It seems we’ve (or I’ve) been assuming that the document root element would be polyfilled in all cases, no matter what selector matches it.

Needs to be fixed in `vaadin-lumo-styles`, and how the dark theme is applied on the global level, to support IE11. This could result in a breaking change, or at least deprecating the current way of applying the dark palette for the global scope.

**Comment by jouni** [comment]
Please follow the issue I just created for `vaadin-lumo-styles`.
